src/components/templates/PartyMap/components/Map/Map.tsx

```diff
@@ -1,199 +1,181 @@
 import React, { useCallback, useEffect, useMemo, useState } from "react";
-import firebase from "firebase/app";
+import { FirebaseReducer } from "react-redux-firebase";
 
-import { User } from "types/User";
-import { PartyMapVenue } from "types/PartyMapVenue";
-import { PartyMapRoomData } from "types/PartyMapRoomData";
+import { User, UserExperienceData } from "types/User";
+import { Room } from "types/rooms";
+import { PartyMapVenue } from "types/venues";
 
-import { enterLocation } from "utils/useLocationUpdateEffect";
-import { getCurrentTimeInUnixEpochSeconds } from "utils/time";
+import { makeUpdateUserGridLocation } from "api/profile";
+
+import { hasElements, isTruthy } from "utils/types";
+import { makeRoomHitFilter } from "utils/filter";
 import { WithId } from "utils/id";
-import { orderedVenuesSelector } from "utils/selectors";
-import { openRoomUrl } from "utils/url";
 
-import { useUser } from "hooks/useUser";
-import { useSelector } from "hooks/useSelector";
 import { useKeyboardControls } from "hooks/useKeyboardControls";
-import { usePartygoers } from "hooks/users";
 
-import Sidebar from "components/molecules/Sidebar";
+// @debt refactor these hooks into somewhere more sensible
+import { useMapGrid } from "./hooks/useMapGrid";
+import { usePartygoersbySeat } from "./hooks/usePartygoersBySeat";
+import { usePartygoersOverlay } from "./hooks/usePartygoersOverlay";
```

Comment:
> `@debt` refactor these hooks into somewhere more sensible

This comment was made when we were going to be sharing them among multiple maps, but I'm not sure that that is the case anymore, so might be fine where they are?